integration_tests/src/main/python/data_gen.py

```diff
@@ -541,6 +541,35 @@ def make_tuple():
     def contains_ts(self):
         return any(child[1].contains_ts() for child in self.children)
 
+class StructGen2(DataGen):
+    """Generate a Struct"""
+    def __init__(self, child1, child2, nullable=True, special_cases=[]):
+        tmp = [StructField('a', child1.data_type, nullable=child1.nullable),
+                StructField('b', child2.data_type, nullable=child2.nullable),
+                StructField('c', child2.data_type, nullable=child2.nullable)]
+        super().__init__(StructType(tmp), nullable=nullable, special_cases=special_cases)
+        self.child1 = child1
+        self.child2 = child2
+
+    def __repr__(self):
+        return super().__repr__() + '(' + str(self.child1) + ',' + str(self.child2) + ')'
+
+    def _cache_repr(self):
+        return super()._cache_repr() + '(' + str(self.child1._cache_repr()) + ',' + str(self.child2._cache_repr()) + ')'
+
+    def start(self, rand):
+        self.child1.start(rand)
+        self.child2.start(rand)
+        def make_tuple():
+            tmp_child2_gen = self.child2.gen()
+            data = [self.child1.gen(), tmp_child2_gen, tmp_child2_gen]
+            return tuple(data)
+        self._start(rand, make_tuple)
+
+def three_col_df2(spark, a_gen, b_gen, length=2048, seed=None, num_slices=None):
```

Comment:
This is named really badly, and I don't see why we need it when all you need is to repeat b_gen twice